filesystem/File.php

```diff
@@ -238,7 +238,7 @@ public function exists() {
 	 */
 	public static function find($filename) {
 		// Get the base file if $filename points to a resampled file
-		$filename = preg_replace('/_resampled\/[^-]+-/', '', $filename);
+		$filename = preg_replace('/_resampled\/(.+\/|[^-]+-)/', '', $filename);
```

Comment:
Doof... I'm wrong again. Ok, on second look that will probably match the old pattern too. :) This is why I hate reviewing regexps! :D

Yeah, a static method with a test would help.